programs/sll_partition.py

```diff
@@ -0,0 +1,34 @@
+# in a sll make a partition to around a value x. All nodes less than x  come before that all nodes equal or greater to x
+# the x value can appear any where in the right partition. x not needs to appear in the middle, just in right partition.
+
+from single_linked_lists import Node
+
+
+def partition(h: Node, x: int):
+    less_nodes = Node(0)
+    head_less = less_nodes
+    great_nodes = Node(0)
+    head_great = great_nodes
+
+    while h:
+        if h.data < x:
+            less_nodes.next = Node(h.data)
```

Comment:
With this line, you are making a copy of each node instead of using the existing nodes. Therefore, your solution would be O(n) instead of O(1) in memory space.
```suggestion
            less_nodes.next = h
```